src/modules/privacy-idea-administration/privacy-idea-administration.service.spec.ts

```diff
@@ -832,4 +837,93 @@ describe(`PrivacyIdeaAdministrationService`, () => {
             expect(result).toBe(requires2fa);
         });
     });
+
+    describe('deleteUser', () => {
+        const personId: string = faker.string.uuid();
+        const referrer: string = faker.string.alpha();
+        const emailAddress: string = faker.internet.email();
+        const event: PersonDeletedEvent = new PersonDeletedEvent(personId, referrer, emailAddress);
+
+        beforeEach(() => {
+            jest.spyOn(
+                service as unknown as { getUserTokens: () => Promise<PrivacyIdeaToken[]> },
+                'getUserTokens',
+            ).mockResolvedValueOnce([]);
+        });
+
+        it(`should delete user`, async () => {
+            httpServiceMock.post.mockReturnValueOnce(mockJWTTokenResponse());
+            httpServiceMock.delete.mockReturnValueOnce(mockEmptyPostResponse());
+            await service.handlePersonDeletedEvent(event);
+        });
+
+        it(`should throw an error if the delete user causes error throw`, async () => {
+            httpServiceMock.post.mockReturnValueOnce(mockJWTTokenResponse());
+            httpServiceMock.delete.mockImplementationOnce(mockErrorResponse);
+
+            await expect(service.handlePersonDeletedEvent(event)).rejects.toThrow(
+                `Error deleting privacyIDEA user: ${mockErrorMsg}`,
+            );
+        });
+
+        it(`should throw an error if the delete user request causes non error throw`, async () => {
+            httpServiceMock.post.mockReturnValueOnce(mockJWTTokenResponse());
+            httpServiceMock.delete.mockImplementationOnce(mockNonErrorThrow);
+
+            await expect(service.handlePersonDeletedEvent(event)).rejects.toThrow(
+                `Error deleting privacyIDEA user: Unknown error occurred`,
+            );
+        });
+    });
+
+    describe('handlePersonDeletedEvent', () => {
+        let personId: string;
+        let referrer: string;
+        let emailAddress: string;
+        let event: PersonDeletedEvent;
+        const mockResetTokenResponse: ResetTokenResponse = createMock<ResetTokenResponse>();
+
+        beforeEach(() => {
+            personId = faker.string.uuid();
+            referrer = faker.string.alpha();
+            emailAddress = faker.internet.email();
+            event = new PersonDeletedEvent(personId, referrer, emailAddress);
+            jest.spyOn(
+                service as unknown as { resetToken: (username: string) => Promise<ResetTokenResponse> },
+                'resetToken',
+            ).mockResolvedValueOnce(mockResetTokenResponse);
```

Comment:
There are a lot of `spyOn`s here, that aren't actually used.
`deleteUser` should not need any spy, since it doesn't call function of the service.

And for `handlePersonDeletedEvent`: I get why you use the spyOn, but I'd recommend moving `handlePersonDeletedEvent` into its own file (I usually separate event handlers from other logic). That way you could use our usual mocking of dependencies instead of spying on the system under test